master/internal/api_project.go

```diff
@@ -214,7 +214,32 @@ func (a *apiServer) getProjectColumnsByID(
 	}
 
 	summaryMetricsSet := make(map[string]struct{})
+	validationMetricsSet := make(map[string]struct{})
 	for _, trial := range summaryMetrics {
+		if rawMetrics, ok := trial.SummaryMetrics["validation_metrics"]; ok {
```

Comment:
nit https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow

Reply:
I'm not sure about this... Because we are reading both `validation_metrics` and `avg_metrics`, can we always assume when `validation_metrics` does not exist, `avg_metrics` also not exist? Otherwise we cannot continue to next iteration just knowing `validation_metrics ` not exist?

Reply:
yeah, i see what you mean. tbh, i'm not anti-goto for exactly this case, but in this case is makes just as much sense to pull out some functions.